File: Easy/0169_majority-element/Solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int majority = nums.length/2;

        for (int i = 0; i < nums.length; i++){
            int count = 0;
            int chose = nums[i];

            for (int j = 0; j < nums.length; j++){
                if (chose == nums[j]){
                    count++;
                }

                if (count > majority){
                    return chose;
                }
            }
        }

        return -1;
    }
}

File: Easy/0169_majority-element/Solution.ts
function majorityElement(nums: number[]): number {
    let candidate = nums[0];
    let count = 0;

    for (const num of nums) {
        if (count === 0) {
            candidate = num;
        }

        if (candidate === num) {
            count++;
        } else {
            count--;
        }
    }

    return candidate;
}
